### PR TITLE
Switch to multi-arch compatible image registry

### PR DESCRIPTION
**Pull request by jakolehm**
- moves all core & addon images to a central registry
- uses multi-arch images wherever possible -> makes possible to mix amd64 & arm64 machines in a cluster
- improves air-gapped setups

### CODE REVIEW

**Comment by jakolehm** [comment]
@jnummelin @kke PTAL

File: addons/ingress-nginx/addon.rb [hunk 1]
```diff
@@ -22,24 +24,10 @@
     apply_resources(
       configmap: config.configmap || {},
       node_selector: config.node_selector || {},
-      image: image_name,
```
Comment (jnummelin) [inline addons/ingress-nginx/addon.rb]: This essentially makes the default backend image non-configurable. Why not just pass the `config.default_backend[:image]`, that should default correctly to `registry.pharos.sh/...`

File: lib/pharos/phases/upgrade_master.rb [hunk 2]
```diff
@@ -36,51 +36,12 @@ def upgrade
...
 
-        dns_patch_thread = create_dns_patch_thread if dns_patch_needed?
         @ssh.tempfile(content: cfg.to_yaml, prefix: "kubeadm.cfg") do |tmp_file|
           @ssh.exec!("sudo /usr/local/bin/pharos-kubeadm-#{Pharos::KUBEADM_VERSION} upgrade apply #{Pharos::KUBE_VERSION} -y --ignore-preflight-errors=all --allow-experimental-upgrades --config #{tmp_file}")
         end
-        dns_patch_thread.join if dns_patch_needed?
 
         logger.info { "Control plane upgrade succeeded!" }
       end
-
```
Comment (jnummelin) [inline lib/pharos/phases/upgrade_master.rb]: Yay, we can finally get rid of this ugly hack :D

File: lib/pharos_cluster.rb [hunk 3]
```diff
@@ -14,4 +14,5 @@ module Pharos
   ETCD_VERSION = ENV.fetch('ETCD_VERSION') { '3.2.18' }
   KUBELET_PROXY_VERSION = '0.3.7'
   COREDNS_VERSION = '1.1.3'
+  TELEMETRY_VERSION = '0.1.0'
```
Comment (jnummelin) [inline lib/pharos_cluster.rb]:
```
$ docker pull registry.pharos.sh/kontenapharos/pharos-telemetry:0.1.0
Error response from daemon: manifest for registry.pharos.sh/kontenapharos/pharos-telemetry:0.1.0 not found
```

File: addons/ingress-nginx/addon.rb (hunk 1 above)
Comment (jakolehm) [reply addons/ingress-nginx/addon.rb]: Good catch, fixed.

File: lib/pharos_cluster.rb (hunk 3 above)
Comment (jakolehm) [reply lib/pharos_cluster.rb]: Multi-arch repo should be there now.

**Pull request merged**